configs/components/yaml-cpp.rb

```diff
@@ -23,9 +23,14 @@
   elsif platform.is_macos?
     cmake_toolchain_file = ""
     cmake = "/usr/local/bin/cmake"
-    if platform.is_cross_compiled?
+    if platform.architecture == 'arm64'
```

Comment:
I think you will still need the `platform.is_cross_compiled?` condition here because the conditions for `osx-11` and `osx-12` won't be executed for x86_64.

Reply:
Hello,
I verified on this one that architecture will be 'arm64' though we are building on x86_64 as we mention the build tatget to be `osx-12-arm64` for particular project. Please let me know in case of any concern I missed here.